modules/apps/push-notifications/push-notifications-sender/push-notifications-sender-apple/src/main/java/com/liferay/push/notifications/sender/apple/internal/ApplePushNotificationsSender.java

```diff
@@ -108,106 +130,115 @@ protected void activate(Map<String, Object> properties) {
 					"Unable to find Apple certificate at " + certificatePath);
 			}
 
-			appleServiceBuilder.withCert(inputStream, certificatePassword);
+			apnsClientBuilder.setClientCredentials(
+				inputStream, certificatePassword);
 		}
 		catch (IOException ioException) {
 			if (_log.isWarnEnabled()) {
 				_log.warn(ioException, ioException);
 			}
 		}
 
-		appleServiceBuilder.withDelegate(new AppleDelegate(_messageBus));
-
 		if (applePushNotificationsSenderConfiguration.sandbox()) {
-			appleServiceBuilder.withSandboxDestination();
+			apnsClientBuilder.setApnsServer(
+				ApnsClientBuilder.DEVELOPMENT_APNS_HOST);
 		}
 		else {
-			appleServiceBuilder.withProductionDestination();
+			apnsClientBuilder.setApnsServer(
+				ApnsClientBuilder.PRODUCTION_APNS_HOST);
 		}
 
-		_apnsService = appleServiceBuilder.build();
+		_apnsClient = apnsClientBuilder.build();
 	}
 
 	protected String buildPayload(JSONObject payloadJSONObject) {
-		PayloadBuilder builder = PayloadBuilder.newPayload();
+		SimpleApnsPayloadBuilder builder = new SimpleApnsPayloadBuilder();
 
 		String title = payloadJSONObject.getString(
```

Comment:
It's possible that you have to sort the builder

Reply:
What do you mean by "sort the builder"? Set the attributes alphabetically?

Reply:
Yes, the attributes that you are setting like builder.setTitle should be called alfabetically.

Reply:
Done!